mitm_proxy/src/mitm_proxy.rs

```diff
@@ -281,18 +290,18 @@ impl MitmProxy {
                 self.requests.push(request);
             }
         }
-
         if let Some(i) = self.state.selected_request {
             ui.columns(2, |columns| {
-                ScrollArea::both()
-                    .id_source("requests_table")
-                    .show(&mut columns[0], |ui| self.table_ui(ui));
-
+                // needs to run before call to table_ui() because table_ui() might change self.state.selected_request
+                // if request is removed
                 ScrollArea::vertical()
                     .id_source("request_details")
                     .show(&mut columns[1], |ui| {
                         self.render_right_panel(ui, i);
                     });
+                ScrollArea::both()
+                    .id_source("requests_table")
+                    .show(&mut columns[0], |ui| self.table_ui(ui));
```

Comment:
To avoid this problem and to avoid the initial problem of considering splitted two parts that are naturally indented (`ScrollArea::both()` is the intrinsic parent of the `Scrollarea::vertical()`) we can refactor this part moving the right panel inside the left panel as his natural child. In this way we can close the left panel before removing the `requests` item.

this is my initial fault and I apologise for this monstrosity.

Reply:
I'm not entirely sure what you mean. 
> `ScrollArea::both()` is the intrinsic parent of the `Scrollarea::vertical()`

You mean this is the case in general? I thought they just create differently scroll-able areas.Or do you mean there is a relationship in this specific case here?

> We can refactor this part moving the right panel inside the left panel as his natural child

So the right "request, response, details" panel should be made the child of the left "table" panel?
I'm not sure how to accomplish that.

Reply:
yes, that is what I meant but I found another simpler solution, I added a easy check before rendering the right panel and this solve the problem in a more idiomatic way